core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/GCSBatchSink.java

```diff
@@ -0,0 +1,151 @@
+  public static final String BUCKET_DES = "GCS bucket to use to store the data";
+  public static final String PROJECT_ID_DES = "Google Cloud Project ID with access to configured GCS buckets";
+  public static final String SERVICE_KEY_FILE_DES = "The Json_Key_File certificate file of the " +
+    "service account used for GCS access";
+  private static final String FILESYSTEM_PROPERTIES_DESCRIPTION = "A JSON string representing a map of properties " +
+    "needed for the distributed file system.";
+  private static final Gson GSON = new Gson();
+
+  private static final Type MAP_STRING_STRING_TYPE = new TypeToken<Map<String, String>>() { }.getType();
+  private static final Logger LOG = LoggerFactory.getLogger(GCSBatchSink.class);
+
+  private final GCSSinkConfig config;
+  protected GCSBatchSink(GCSSinkConfig config) {
+    super(config);
+    this.config = config;
+    if (!this.config.containsMacro("fileSystemProperties") && !this.config.containsMacro("Bucket_Key") &&
+      !this.config.containsMacro("Project_Id")) {
+      this.config.fileSystemProperties = updateFileSystemProperties(this.config.fileSystemProperties,
+                                                                    this.config.projectId,
+                                                                    this.config.jsonKey);
+    }
+  }
+
+  @Override
+  public final void prepareRun(BatchSinkContext context) {
+    OutputFormatProvider outputFormatProvider = createOutputFormatProvider(context);
+    Map<String, String> outputConfig = new HashMap<>(outputFormatProvider.getOutputFormatConfiguration());
+    if (config.fileSystemProperties != null) {
+      Map<String, String> properties = GSON.fromJson(config.fileSystemProperties, MAP_STRING_STRING_TYPE);
+      outputConfig.putAll(properties);
+    }
+    LOG.debug("test the outputconfig {}", outputConfig);
+    context.addOutput(Output.of(config.referenceName, new SinkOutputFormatProvider(
+      outputFormatProvider.getOutputFormatClassName(), outputConfig)));
+  }
+
+  protected abstract OutputFormatProvider createOutputFormatProvider(BatchSinkContext context);
+
+  private static String updateFileSystemProperties(@Nullable String fileSystemProperties,
+                                                   String projectId, String jsonKey) {
+    Map<String, String> providedProperties;
+    if (fileSystemProperties == null) {
+      providedProperties = new HashMap<>();
+    } else {
+      providedProperties = GSON.fromJson(fileSystemProperties, MAP_STRING_STRING_TYPE);
+    }
+    providedProperties.put("fs.gs.project.id", projectId);
+    providedProperties.put("google.cloud.auth.service.account.json.keyfile", jsonKey);
+    return GSON.toJson(providedProperties);
+  }
+
+  /**
+   * GCS Sink configuration.
+   */
+  public static class GCSSinkConfig extends ReferencePluginConfig {
+
+    @Name("Bucket_Key")
+    @Description(BUCKET_DES)
+    @Macro
+    protected String bucketKey;
+
+    @Name("Project_Id")
```

Comment:
Remove underscores here and in the json config. Make sure naming is using Title Case.